### PR TITLE
Metadata unification

### LINKED ISSUE #195: Refactor client side minerva metadata
There are a few methods used by minerva models that should be unified into a `metadata` method.

  * mM
  * setMinervaMetadata
  * getMinervaMetadata

See [this comment](https://github.com/Kitware/minerva/pull/174#discussion_r44328436) and those above it.

### PR DESCRIPTION
**Pull request by mgrauer**
Fixes #195.

Based on top of commits from #174.

ping @danlamanna

### CODE REVIEW

File: server/jobs/bsve_search_worker.py [hunk 1]
```diff
@@ -88,12 +86,42 @@ def run(job):
...
+            minerva_metadata['geojson_file'] = {
+                '_id': existing['_id'],
+                'name': geojsonFilename
+            }
+        else:
+            raise (Exception('Cannot find file %s in dataset %s' %
+                   (geojsonFilename, datasetId)))
+
+        mM(dataset, minerva_metadata)
         job_model.updateJob(job, status=JobStatus.SUCCESS)
```
Comment (danlamanna) [inline server/jobs/bsve_search_worker.py]: I know calling it just ```metadata``` here would be ambiguous, but can we find some way to rename ```mM``` server-side too?

Comment (mgrauer) [inline server/jobs/bsve_search_worker.py]: Opened issue #198.

**Comment by mgrauer** [comment]
Was that your only comment @danlamanna , or did your QA get interrupted?  Just wondering where this PR stands.

**Comment by danlamanna** [comment]
Yes, otherwise LGTM.

**Pull request merged**